testing/mock_canvas.cc

```diff
@@ -0,0 +1,283 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/testing/mock_canvas.h"
+
+#include "flutter/fml/logging.h"
+#include "third_party/skia/include/core/SkImageInfo.h"
+
+namespace flutter {
+namespace testing {
+
+MockCanvas::MockCanvas()
+    : SkCanvasVirtualEnforcer<SkCanvas>(64, 64),
```

Comment:
This just creates a NoPixelDevice. You should just be able to use the default constructor. The 64 is a bit confusing and I initially thought it might be a cull.